src/cascadia/TerminalSettingsEditor/Actions.h

```diff
@@ -59,20 +62,29 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         void AttemptAcceptChanges(hstring newKeyChordText);
         void DeleteKeyBinding() { _DeleteKeyBindingRequestedHandlers(*this, _Keys); }
 
-        VIEW_MODEL_OBSERVABLE_PROPERTY(bool, IsInEditMode, false);
+        // ProposedAction:   the entry selected by the combo box; may disagree with the settings model.
+        // CurrentAction:    the combo box item that maps to the settings model value.

```

Comment:
Really it's because we have the "cancel"/"ok" system (aka the "edit mode" system). `CurrentAction` serves as...
1. a record of what to set `ProposedAction` to on a cancellation
2. a form of translation between `ProposedAction` and the settings model

If we got rid of the "edit mode" system, we don't have to worry about (1) anymore. To get rid of (2), `KeyBindingViewModel` would need a reference to the `ActionMap`. But even then, I think it's better to keep the `ActionMap` reference at a higher layer so that we can identify conflicting key bindings and key chords before editing the settings model. I think that's a better separation of responsibilities imo.